kuronometer-core/src/main/scala/com/github/pedrovgs/kuronometer/Kuronometer.scala

```diff
@@ -0,0 +1,92 @@
+package com.github.pedrovgs.kuronometer
+
+import cats.free.Free
+import com.github.pedrovgs.kuronometer.KuronometerResults.{ConnectionError, KuronometerResult}
+import com.github.pedrovgs.kuronometer.app.KuronometerProgram
+import com.github.pedrovgs.kuronometer.free.algebra.{ReporterOps, ViewOps}
+import com.github.pedrovgs.kuronometer.free.domain.View.Message
+import com.github.pedrovgs.kuronometer.free.domain._
+import com.github.pedrovgs.kuronometer.free.interpreter.Interpreters
+import com.github.pedrovgs.kuronometer.free.interpreter.formatter.DurationFormatter.NanosecondsFormat.format
+import com.github.pedrovgs.kuronometer.free.interpreter.formatter.SummaryBuildStageExecutionFormatter
+
+class Kuronometer(implicit R: ReporterOps[KuronometerProgram], V: ViewOps[KuronometerProgram], interpreters: Interpreters) {
+
+  private val kuronometerHeader = "== Kuronometer =="
+
+  def reportBuildFinished(buildExecution: BuildExecution, config: Config): KuronometerResult[BuildExecution] = {
+    val filteredBuildExecution = filterBuildExecutionData(buildExecution, config)
+    val program = for {
+      result <- reportBuildExecution(filteredBuildExecution, config)
+      _ <- showReportResult(result, config)
+    } yield result
+    program.foldMap(interpreters.kuronometerInterpreter)
+  }
+
+  def getTotalBuildExecutionSummary: KuronometerResult[SummaryBuildStagesExecution] = {
+    val program = for {
+      totalSummary <- R.getTotalBuildExecution
+      _ <- showBuildExecutionSummary(totalSummary)
+    } yield totalSummary
+    program.foldMap(interpreters.kuronometerInterpreter)
+  }
+
+  def getTodayBuildExecutionSummary: KuronometerResult[SummaryBuildStagesExecution] = {
+    val program = for {
+      todaySummary <- R.getTodayBuildExecution
+      _ <- showBuildExecutionSummary(todaySummary)
+    } yield todaySummary
+    program.foldMap(interpreters.kuronometerInterpreter)
+  }
+
+  private def reportBuildExecution(buildExecution: BuildExecution, config: Config): Free[KuronometerProgram, KuronometerResult[BuildExecution]] = {
+    for {
```

Comment:
There is no reason to use for comprehensions here since it's a single operation. This can be implemented simply with `Free#map`